src/bosh_azure_cpi/lib/cloud/azure/storage/blob_manager.rb

```diff
@@ -417,12 +417,19 @@ def _copy_blob_from_uri(container_name, blob_name, source_blob_uri, options)
     rescue StandardError => e
       # create the container if it does not exists
       if e.inspect.include?('ContainerNotFound')
-        @blob_service_client.create_container(container_name, merge_storage_common_options)
+        _create_container(container_name)
         @blob_service_client.copy_blob_from_uri(container_name, blob_name, source_blob_uri, options)
```

Comment:
this means that ` @blob_service_client.copy_blob_from_uri(container_name, blob_name, source_blob_uri, options)` will be called in two different process, can lead to another race error or even message the dest blob.
suggest using a lock to avoid race.

Reply:
yes, a lock is needed, not only here, but also many other place, since the change would be merged to master, and it's near the release date. so let's only fix this issue will minor change.

Reply:
I think, https://github.com/cloudfoundry/bosh-azure-cpi-release/blob/09f6a50fc4a009454a4a79bdeead9eed7c4da802/src/bosh_azure_cpi/lib/cloud/azure/stemcell/stemcell_manager.rb#L91-L92 has guaranteed that only one process is copying the stemcell. No?

Reply:
@bingosummer  cool, then we do not need this change for the master branch for now.
I found this issue for the config disk container. switched to the develop branch now.